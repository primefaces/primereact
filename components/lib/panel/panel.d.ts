/**
 *
 * Panel is a grouping component providing with content toggle feature.
 *
 * [Live Demo](https://www.primereact.org/panel/)
 *
 * @module panel
 *
 */
import * as React from 'react';
import { CSSTransitionProps } from '../csstransition';
import { IconType } from '../utils';

/**
 * Custom panel header template options.
 */
interface PanelHeaderTemplateOptions {
    /**
     * Style class of the panel.
     */
    className: string;
    /**
     * Style class of the panel title.
     */
    titleClassName: string;
    /**
     * Style class of the panel icons.
     */
    iconsClassName: string;
    /**
     * Style class of the panel toggler.
     */
    togglerClassName: string;
    /**
     * Style class of the panel toggler icon.
     */
    togglerIconClassName: string;
    /**
     * Callback to invoke when the toggler button is clicked.
     * @param {React.MouseEvent<HTMLElement>} event Browser event.
     */
    onTogglerClick(event: React.MouseEvent<HTMLElement>): void;
    /**
     * The JSX element that represents the title of the panel.
     */
    titleElement: JSX.Element;
    /**
     * The JSX element that represents the icons of the panel.
     */
    iconsElement: JSX.Element;
    /**
     * The JSX element that represents the toggler of the panel.
     */
    togglerElement: JSX.Element;
    /**
     * The JSX element that represents the panel.
     */
    element: JSX.Element;
    /**
     * The props of the Panel component.
     */
    props: PanelProps;
    /**
     * Whether the panel header is collapsed or not.
     */
    collapsed: boolean;
}

/**
 * Custom toggle event.
 * @see {@link PanelProps.onToggle}
 * @event
 */
interface PanelToggleEvent {
    /**
     * Browser event.
     */
    originalEvent: React.MouseEvent<HTMLElement>;
    /**
     * Collapsed state as a boolean.
     */
    value: boolean;
}

/**
 * Defines valid properties in Panel component. In addition to these, all properties of HTMLDivElement can be used in this component.
 * @group Properties
 */
export interface PanelProps extends Omit<React.DetailedHTMLProps<React.HTMLAttributes<HTMLDivElement>, HTMLDivElement>, 'ref'> {
    /**
     * Custom header template of the panel.
     */
    header?: React.ReactNode | undefined;
    /**
     * Header template of the panel to customize more.
     * @param {PanelHeaderTemplateOptions} options - Options to customize the header template.
     */
    headerTemplate?: React.ReactNode | ((options: PanelHeaderTemplateOptions) => React.ReactNode);
    /**
     * Defines if content of panel can be expanded and collapsed.
     * @defaultValue false
     */
    toggleable?: boolean | undefined;
    /**
     * Defines the initial state of panel content, supports one or two-way binding as well.
     * @defaultValue false
     */
    collapsed?: boolean | undefined;
    /**
     * Icon of a expanded tab.
     */
    expandIcon?: IconType<PanelProps> | undefined;
    /**
     * Icon of a collapsed tab.
     */
    collapseIcon?: IconType<PanelProps> | undefined;
    /**
     * Custom icons template for the header.
     */
    icons?: React.ReactNode | ((props: PanelProps) => React.ReactNode);
    /**
     * The properties of CSSTransition can be customized, except for "nodeRef" and "in" properties.
     * @type {CSSTransitionProps}
     */
    transitionOptions?: CSSTransitionProps | undefined;
    /**
     * Callback to invoke when a tab gets expanded.
     * @param {React.SyntheticEvent} event - Browser event.
     */
    onExpand?(event: React.SyntheticEvent): void;
    /**
     * Callback to invoke when an active tab is collapsed by clicking on the header.
     * @param {React.SyntheticEvent} event - Browser event.
     */
    onCollapse?(event: React.SyntheticEvent): void;
    /**
     * Callback to invoke when a tab gets expanded.
     * @param {PanelToggleEvent} event - Custom toggle event.
     */
    onToggle?(event: PanelToggleEvent): void;
    /**
     * Used to get the child elements of the component.
     * @readonly
     */
    children?: React.ReactNode | undefined;
}

/**
 * **PrimeReact - Panel**
 *
 * _Panel is a grouping component providing with content toggle feature._
 *
 * [Live Demo](https://www.primereact.org/panel/)
 * --- ---
 * ![PrimeReact](https://primefaces.org/cdn/primereact/images/logo-100.png)
 *
 * @group Component
 */
export declare class Panel extends React.Component<PanelProps, any> {
    /**
     * Used to get container element.
     * @return {HTMLDivElement} Container element
     */
    public getElement(): HTMLDivElement;
    /**
     * Used to get content of the panel.
     * @return {HTMLDivElement} Content element
     */
    public getContent(): HTMLDivElement;
}
