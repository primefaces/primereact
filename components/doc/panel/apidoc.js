import Link from 'next/link';
import { DevelopmentSection } from '../common/developmentsection';
import { DocSectionText } from '../common/docsectiontext';
import { DocSubSection } from '../common/docsubsection';

export function ApiDoc(props) {
    return (
        <>
            <DocSectionText {...props}></DocSectionText>
            <DocSubSection id="properties" label="Properties">
                <div className="doc-tablewrapper">
                    <table className="doc-table">
                        <thead>
                            <tr>
                                <th>Name</th>
                                <th>Type</th>
                                <th>Default</th>
                                <th>Description</th>
                            </tr>
                        </thead>
                        <tbody>
                            <tr>
                                <td>id</td>
                                <td>string</td>
                                <td>null</td>
                                <td>Unique identifier of the element.</td>
                            </tr>
                            <tr>
                                <td>header</td>
                                <td>any</td>
                                <td>null</td>
                                <td>Custom header template of the panel.</td>
                            </tr>
                            <tr>
                                <td>headerTemplate</td>
                                <td>any</td>
                                <td>null</td>
                                <td>Header template of the panel to customize more.</td>
                            </tr>
                            <tr>
                                <td>toggleable</td>
                                <td>boolean</td>
                                <td>false</td>
                                <td>Defines if content of panel can be expanded and collapsed.</td>
                            </tr>
                            <tr>
                                <td>icons</td>
                                <td>any</td>
                                <td>null</td>
                                <td>Custom icons template for the header.</td>
                            </tr>
                            <tr>
                                <td>style</td>
                                <td>object</td>
                                <td>null</td>
                                <td>Inline style of the element.</td>
                            </tr>
                            <tr>
                                <td>className</td>
                                <td>string</td>
                                <td>null</td>
                                <td>Style class of the element.</td>
                            </tr>
                            <tr>
                                <td>collapsed</td>
                                <td>boolean</td>
                                <td>false</td>
                                <td>Defines the initial state of panel content, supports one or two-way binding as well.</td>
                            </tr>
                            <tr>
                                <td>transitionOptions</td>
                                <td>object</td>
                                <td>null</td>
                                <td>
                                    The properties of{' '}
                                    <a href="https://reactcommunity.org/react-transition-group/css-transition" rel="noopener noreferrer" target="_blank">
                                        CSSTransition
                                    </a>{' '}
                                    can be customized, except for "nodeRef" and "in" properties.
                                </td>
                            </tr>
                        </tbody>
                    </table>
                </div>
            </DocSubSection>

            <DocSubSection id="events" label="Events">
                <div className="doc-tablewrapper">
                    <table className="doc-table">
                        <thead>
                            <tr>
                                <th>Name</th>
                                <th>Parameters</th>
                                <th>Description</th>
                            </tr>
                        </thead>
                        <tbody>
                            <tr>
                                <td>onCollapse</td>
                                <td>event.originalEvent: browser event </td>
                                <td>Callback to invoke when an active tab is collapsed by clicking on the header.</td>
                            </tr>
                            <tr>
                                <td>onExpand</td>
                                <td>event.originalEvent: browser event </td>
                                <td>Callback to invoke when a tab gets expanded.</td>
                            </tr>
                            <tr>
                                <td>onToggle</td>
                                <td>
                                    event.originalEvent: browser event <br />
                                    event.value: collapsed state as a boolean
                                </td>
                                <td>Callback to invoke when a tab gets expanded.</td>
                            </tr>
                        </tbody>
                    </table>
                </div>
            </DocSubSection>
        </>
    );
}
