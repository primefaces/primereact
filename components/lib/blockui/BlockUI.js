import * as React from 'react';
import PrimeReact from '../api/Api';
import { useMountEffect, useUnmountEffect, useUpdateEffect } from '../hooks/Hooks';
import { Portal } from '../portal/Portal';
import { classNames, DomHandler, ObjectUtils, ZIndexUtils } from '../utils/Utils';
import { BlockUIBase } from './BlockUIBase';

export const BlockUI = React.forwardRef((inProps, ref) => {
    const props = BlockUIBase.getProps(inProps);

    const [visibleState, setVisibleState] = React.useState(props.blocked);
    const elementRef = React.useRef(null);
    const maskRef = React.useRef(null);

    const block = () => {
        setVisibleState(true);
    };

    const unblock = () => {
        const callback = () => {
            setVisibleState(false);

            props.fullScreen && DomHandler.removeClass(document.body, 'p-overflow-hidden');
            props.onUnblocked && props.onUnblocked();
        };

        if (maskRef.current) {
            DomHandler.addClass(maskRef.current, 'p-component-overlay-leave');
            maskRef.current.addEventListener('animationend', () => {
                ZIndexUtils.clear(maskRef.current);
                callback();
            });
        } else {
            callback();
        }
    };

    const onPortalMounted = () => {
        if (props.fullScreen) {
            DomHandler.addClass(document.body, 'p-overflow-hidden');
            document.activeElement.blur();
        }

        if (props.autoZIndex) {
            const key = props.fullScreen ? 'modal' : 'overlay';

            ZIndexUtils.set(key, maskRef.current, PrimeReact.autoZIndex, props.baseZIndex || PrimeReact.zIndex[key]);
        }

        props.onBlocked && props.onBlocked();
    };

    useMountEffect(() => {
        visibleState && block();
    });

    useUpdateEffect(() => {
        props.blocked ? block() : unblock();
    }, [props.blocked]);

    useUnmountEffect(() => {
        if (props.fullScreen) {
            DomHandler.removeClass(document.body, 'p-overflow-hidden');
        }

        ZIndexUtils.clear(maskRef.current);
    });

    React.useImperativeHandle(ref, () => ({
        props,
        block,
        unblock,
        getElement: () => elementRef.current
    }));

    const createMask = () => {
        if (visibleState) {
            const appendTo = props.fullScreen ? document.body : 'self';
            const className = classNames(
                'p-blockui p-component-overlay p-component-overlay-enter',
                {
                    'p-blockui-document': props.fullScreen
                },
                props.className
            );
            const content = props.template ? ObjectUtils.getJSXElement(props.template, props) : null;
            const mask = (
                <div ref={maskRef} className={className} style={props.style}>
                    {content}
                </div>
            );

            return <Portal element={mask} appendTo={appendTo} onMounted={onPortalMounted} />;
        }

        return null;
    };

    const otherProps = BlockUIBase.getOtherProps(props);
    const mask = createMask();
    const className = classNames('p-blockui-container', props.containerClassName);

    return (
        <div id={props.id} ref={elementRef} className={className} style={props.containerStyle} {...otherProps}>
            {props.children}
            {mask}
        </div>
    );
});

BlockUI.displayName = 'BlockUI';
