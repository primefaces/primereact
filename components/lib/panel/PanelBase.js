import { ObjectUtils } from '../utils/Utils';

export const PanelBase = {
    defaultProps: {
        __TYPE: 'Panel',
        id: null,
        header: null,
        headerTemplate: null,
        toggleable: null,
        style: null,
        className: null,
        collapsed: null,
        expandIcon: 'pi pi-plus',
        collapseIcon: 'pi pi-minus',
        icons: null,
        transitionOptions: null,
        onExpand: null,
        onCollapse: null,
        onToggle: null,
        children: undefined
    },
    getProps: (props) => ObjectUtils.getMergedProps(props, PanelBase.defaultProps),
    getOtherProps: (props) => ObjectUtils.getDiffProps(props, PanelBase.defaultProps)
};
