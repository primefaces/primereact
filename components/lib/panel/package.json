{
    "main": "./panel.cjs.js",
    "module": "./panel.esm.js",
    "unpkg": "./panel.min.js",
    "types": "./panel.d.ts",
    "sideEffects": false
}
