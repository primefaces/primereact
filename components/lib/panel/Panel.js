import * as React from 'react';
import { CSSTransition } from '../csstransition/CSSTransition';
import { useMountEffect } from '../hooks/Hooks';
import { Ripple } from '../ripple/Ripple';
import { classNames, IconUtils, ObjectUtils, UniqueComponentId } from '../utils/Utils';
import { PanelBase } from './PanelBase';

export const Panel = React.forwardRef((inProps, ref) => {
    const props = PanelBase.getProps(inProps);

    const [idState, setIdState] = React.useState(props.id);
    const [collapsedState, setCollapsedState] = React.useState(props.collapsed);
    const elementRef = React.useRef(ref);
    const contentRef = React.useRef(null);
    const collapsed = props.toggleable ? (props.onToggle ? props.collapsed : collapsedState) : false;
    const headerId = idState + '_header';
    const contentId = idState + '_content';

    const toggle = (event) => {
        if (props.toggleable) {
            collapsed ? expand(event) : collapse(event);

            if (props.onToggle) {
                props.onToggle({
                    originalEvent: event,
                    value: !collapsed
                });
            }
        }

        event.preventDefault();
    };

    const expand = (event) => {
        if (!props.onToggle) {
            setCollapsedState(false);
        }

        props.onExpand && props.onExpand(event);
    };

    const collapse = (event) => {
        if (!props.onToggle) {
            setCollapsedState(true);
        }

        props.onCollapse && props.onCollapse(event);
    };

    React.useImperativeHandle(ref, () => ({
        props,
        getElement: () => elementRef.current,
        getContent: () => contentRef.current
    }));

    React.useEffect(() => {
        ObjectUtils.combinedRefs(elementRef, ref);
    }, [elementRef, ref]);

    useMountEffect(() => {
        if (!idState) {
            setIdState(UniqueComponentId());
        }
    });

    const createToggleIcon = () => {
        if (props.toggleable) {
            const buttonId = idState + '_label';
            const toggleIcon = collapsed ? props.expandIcon : props.collapseIcon;

            return (
                <button className="p-panel-header-icon p-panel-toggler p-link" onClick={toggle} id={buttonId} aria-controls={contentId} aria-expanded={!collapsed} role="tab">
                    {IconUtils.getJSXIcon(toggleIcon, undefined, { props, collapsed })}
                    <Ripple />
                </button>
            );
        }

        return null;
    };

    const createHeader = () => {
        const header = ObjectUtils.getJSXElement(props.header, props);
        const icons = ObjectUtils.getJSXElement(props.icons, props);
        const togglerElement = createToggleIcon();
        const titleElement = (
            <span className="p-panel-title" id={headerId}>
                {header}
            </span>
        );
        const iconsElement = (
            <div className="p-panel-icons">
                {icons}
                {togglerElement}
            </div>
        );
        const content = (
            <div className="p-panel-header">
                {titleElement}
                {iconsElement}
            </div>
        );

        if (props.headerTemplate) {
            const defaultContentOptions = {
                className: 'p-panel-header',
                titleClassName: 'p-panel-title',
                iconsClassName: 'p-panel-icons',
                togglerClassName: 'p-panel-header-icon p-panel-toggler p-link',
                togglerIconClassName: collapsed ? props.expandIcon : props.collapseIcon,
                onTogglerClick: toggle,
                titleElement,
                iconsElement,
                togglerElement,
                element: content,
                props,
                collapsed
            };

            return ObjectUtils.getJSXElement(props.headerTemplate, defaultContentOptions);
        } else if (props.header || props.toggleable) {
            return content;
        }

        return null;
    };

    const createContent = () => {
        return (
            <CSSTransition nodeRef={contentRef} classNames="p-toggleable-content" timeout={{ enter: 1000, exit: 450 }} in={!collapsed} unmountOnExit options={props.transitionOptions}>
                <div ref={contentRef} className="p-toggleable-content" aria-hidden={collapsed} role="region" id={contentId} aria-labelledby={headerId}>
                    <div className="p-panel-content">{props.children}</div>
                </div>
            </CSSTransition>
        );
    };

    const otherProps = PanelBase.getOtherProps(props);
    const className = classNames(
        'p-panel p-component',
        {
            'p-panel-toggleable': props.toggleable
        },
        props.className
    );
    const header = createHeader();
    const content = createContent();

    return (
        <div id={props.id} ref={elementRef} className={className} style={props.style} {...otherProps}>
            {header}
            {content}
        </div>
    );
});

Panel.displayName = 'Panel';
