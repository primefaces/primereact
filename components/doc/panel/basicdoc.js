import { Panel } from '../../lib/panel/Panel';
import { DocSectionText } from '../common/docsectiontext';
import { DocSectionCode } from '../common/docsectioncode';

export function BasicDoc(props) {
    const code = {
        basic: `
<Panel header="Header">
    <p className="m-0">
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. 
        Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo
        consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. 
        Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
    </p>
</Panel>
        `,
        javascript: `
import React from 'react'; 
import { Panel } from 'primereact/panel';

export default function BasicDemo() {
    return (
        <Panel header="Header">
            <p className="m-0">
                Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. 
                Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo
                consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. 
                Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
            </p>
        </Panel>
    )
}
        `,
        typescript: `
import React from 'react'; 
import { Panel } from 'primereact/panel';

export default function BasicDemo() {
    return (
        <Panel header="Header">
            <p className="m-0">
                Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. 
                Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo
                consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. 
                Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
            </p>
        </Panel>
    )
}
        `
    };

    return (
        <>
            <DocSectionText {...props}>
                <p>
                    A simple Panel is created with a <i>header</i> property along with the content as children.
                </p>
            </DocSectionText>
            <div className="card">
                <Panel header="Header">
                    <p className="m-0">
                        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo
                        consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
                    </p>
                </Panel>
            </div>
            <DocSectionCode code={code} />
        </>
    );
}
