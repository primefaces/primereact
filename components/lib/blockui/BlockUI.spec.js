import '@testing-library/jest-dom';
import { act, render } from '@testing-library/react';
import * as React from 'react';
import { snapshotParent } from '../../test';
import { Panel } from '../panel/Panel';
import { BlockUI } from './BlockUI';

describe('BlockUI', () => {
    snapshotParent(<BlockUI blocked={true} fullScreen />, 'block fullscreen');
    snapshotParent(
        <BlockUI blocked={true}>
            <Panel>blocked</Panel>
        </BlockUI>,
        'block panel'
    );
    snapshotParent(
        <BlockUI blocked={false}>
            <Panel>unblocked</Panel>
        </BlockUI>,
        'unblock panel'
    );
    test('block and unblock panel events', async () => {
        // Arrange
        const ref = React.createRef();
        const { container } = render(
            <BlockUI ref={ref} blocked={false}>
                <Panel>unblocked</Panel>
            </BlockUI>
        );

        // Act
        act(() => {
            ref.current.block();
            ref.current.getElement();
        });

        // Assert
        expect(container).toMatchSnapshot('blocked-event');

        // Act
        act(() => {
            ref.current.unblock();
        });

        // Assert
        expect(container).toMatchSnapshot('unblocked-event');
    });
    test('block and unblock fullscreen events', async () => {
        // Arrange
        const ref = React.createRef();
        const { container } = render(<BlockUI ref={ref} blocked={false} fullScreen></BlockUI>);

        // Act
        act(() => {
            ref.current.block();
        });

        // Assert
        expect(container.parentElement).toMatchSnapshot('blocked-event-fullscreen');

        // Act
        act(() => {
            ref.current.unblock();
        });

        // Assert
        expect(container.parentElement).toMatchSnapshot('unblocked-event-fullscreen');
    });
    snapshotParent(
        <BlockUI blocked={true} className="block-jest" style={{ cursor: 'move' }} containerClassName="container-jest" containerStyle={{ cursor: 'pointer' }}>
            <Panel>style + class</Panel>
        </BlockUI>,
        'container style and className'
    );
});
