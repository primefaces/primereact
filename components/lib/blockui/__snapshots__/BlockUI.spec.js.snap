// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlockUI block and unblock fullscreen events: blocked-event-fullscreen 1`] = `
<body
  class="p-overflow-hidden"
>
  <div>
    <div
      class="p-blockui-container"
    />
  </div>
  <div
    class="p-blockui p-component-overlay p-component-overlay-enter p-blockui-document"
    style="z-index: 3204;"
  />
</body>
`;

exports[`BlockUI block and unblock fullscreen events: unblocked-event-fullscreen 1`] = `
<body
  class="p-overflow-hidden"
>
  <div>
    <div
      class="p-blockui-container"
    />
  </div>
  <div
    class="p-blockui p-component-overlay p-component-overlay-enter p-blockui-document p-component-overlay-leave"
    style="z-index: 3204;"
  />
</body>
`;

exports[`BlockUI block and unblock panel events: blocked-event 1`] = `
<div>
  <div
    class="p-blockui-container"
  >
    <div
      class="p-panel p-component"
    >
      <div
        aria-hidden="false"
        aria-labelledby="pr_id_3_header"
        class="p-toggleable-content"
        id="pr_id_3_content"
        role="region"
      >
        <div
          class="p-panel-content"
        >
          unblocked
        </div>
      </div>
    </div>
    <div
      class="p-blockui p-component-overlay p-component-overlay-enter"
      style="z-index: 2103;"
    />
  </div>
</div>
`;

exports[`BlockUI block and unblock panel events: unblocked-event 1`] = `
<div>
  <div
    class="p-blockui-container"
  >
    <div
      class="p-panel p-component"
    >
      <div
        aria-hidden="false"
        aria-labelledby="pr_id_3_header"
        class="p-toggleable-content"
        id="pr_id_3_content"
        role="region"
      >
        <div
          class="p-panel-content"
        >
          unblocked
        </div>
      </div>
    </div>
    <div
      class="p-blockui p-component-overlay p-component-overlay-enter p-component-overlay-leave"
      style="z-index: 2103;"
    />
  </div>
</div>
`;

exports[`BlockUI block fullscreen 1`] = `
<body
  class="p-overflow-hidden"
>
  <div>
    <div
      class="p-blockui-container"
    />
  </div>
  <div
    class="p-blockui p-component-overlay p-component-overlay-enter p-blockui-document"
    style="z-index: 1101;"
  />
</body>
`;

exports[`BlockUI block panel 1`] = `
<body
  class=""
>
  <div>
    <div
      class="p-blockui-container"
    >
      <div
        class="p-panel p-component"
      >
        <div
          aria-hidden="false"
          aria-labelledby="pr_id_1_header"
          class="p-toggleable-content"
          id="pr_id_1_content"
          role="region"
        >
          <div
            class="p-panel-content"
          >
            blocked
          </div>
        </div>
      </div>
      <div
        class="p-blockui p-component-overlay p-component-overlay-enter"
        style="z-index: 2102;"
      />
    </div>
  </div>
</body>
`;

exports[`BlockUI container style and className 1`] = `
<body
  class=""
>
  <div>
    <div
      class="p-blockui-container container-jest"
      style="cursor: pointer;"
    >
      <div
        class="p-panel p-component"
      >
        <div
          aria-hidden="false"
          aria-labelledby="pr_id_4_header"
          class="p-toggleable-content"
          id="pr_id_4_content"
          role="region"
        >
          <div
            class="p-panel-content"
          >
            style + class
          </div>
        </div>
      </div>
      <div
        class="p-blockui p-component-overlay p-component-overlay-enter block-jest"
        style="cursor: move; z-index: 4205;"
      />
    </div>
  </div>
</body>
`;

exports[`BlockUI unblock panel 1`] = `
<body
  class=""
>
  <div>
    <div
      class="p-blockui-container"
    >
      <div
        class="p-panel p-component"
      >
        <div
          aria-hidden="false"
          aria-labelledby="pr_id_2_header"
          class="p-toggleable-content"
          id="pr_id_2_content"
          role="region"
        >
          <div
            class="p-panel-content"
          >
            unblocked
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
